### PR TITLE
[REF] mail,hr_holidays: `PartnerImStatusIcon`, review and simplify scss

### PR DESCRIPTION
**Pull request by xlu-odoo**
Part of the overall v16 SCSS optimization/restyle, task-2704984

task-2832067

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss [hunk 1]
```diff
@@ -1,49 +1,12 @@
...
 // ------------------------------------------------------------------
-
+$o-status-icon-size: 1.2em;
 .o_PartnerImStatusIcon {
-    display: flex;
-    flex-flow: column;
-
-    width: 1.2em;
-    height: 1.2em;
-    line-height: 1.3em;
```
Comment (Brieuc-brd) [inline addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss:11]: Are you sure you want to remove the `line-height` property? There's probably a reason why it is used (in a specific situation or for compatibility reasons).

**Review by Brieuc-brd — COMMENTED** [review]
Hello @xlu-odoo 👋 
Some remarks, feel free to ask me if you have any question. 👍

File: addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss [hunk 2]
```diff
@@ -1,49 +1,12 @@
 // ------------------------------------------------------------------
 // Layout
 // ------------------------------------------------------------------
-
+$o-status-icon-size: 1.2em;
```
[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss:4]: Yes, good idea to keep all variables in a single place (and rename so it's consistent and clear what this means in a more global scope).

File: addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.xml [hunk 3]
```diff
@@ -3,11 +3,11 @@
 
     <t t-name="mail.PartnerImStatusIcon" owl="1">
         <t t-if="partner">
-            <span class="o_PartnerImStatusIcon fa-stack"
+            <span class="o_PartnerImStatusIcon d-flex flex-column fa-stack"
```
[the remark this answers is not in the document]
Comment (Brieuc-brd) [reply addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.xml:6]: Ok my bad, the issue doesn't come from here but from `hr_holidays`

[the remark this answers is not in the document]
Comment (xlu-odoo) [reply addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.xml:6]: Updated (w/ https://github.com/odoo/odoo/pull/89600)

File: addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss [hunk 4]
```diff
@@ -1,49 +1,12 @@
 // ------------------------------------------------------------------
 // Layout
 // ------------------------------------------------------------------
-
```
Comment (Brieuc-brd) [inline addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss:4]: Why remove this space? :) I think you should put it back to keep the right structure.

Comment (xlu-odoo) [reply addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.scss:4]: Updated 👍

File: addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.xml [hunk 5]
```diff
@@ -19,19 +19,19 @@
             >
                 <t t-if="props.hasBackground">
                     <i class="o_PartnerImStatusIcon_outerBackground fa fa-circle fa-stack-1x"/>
-                    <i class="o_PartnerImStatusIcon_innerBackground fa fa-circle fa-stack-1x"/>
+                    <i class="o_PartnerImStatusIcon_innerBackground fa fa-circle fa-stack-1x text-white"/>
                 </t>
                 <t t-if="partner.im_status === 'online'">
-                    <i class="o_PartnerImStatusIcon_icon o-online fa fa-circle fa-stack-1x" title="Online" role="img" aria-label="User is online"/>
+                    <i class="o_PartnerImStatusIcon_icon o-online fa fa-circle fa-stack-1x text-primary" title="Online" role="img" aria-label="User is online"/>
```
Comment (Brieuc-brd) [inline addons/mail/static/src/components/partner_im_status_icon/partner_im_status_icon.xml:25]: This works fine on Enterprise but it returns a wrong color on Community (purple).
I don't think there's a class to handle `$o-enterprise-primary-color`. We could use something like `text-success` instead to have a main color for online status between Enterprise and Community, but not in this PR. It's something we should probably discuss with @stefanorigano 🙂

**Comment by Brieuc-brd** [comment]
Ok to me. @xlu-odoo, you can remove the draft state and set your task to « Ready 🟢 ».

@alexkuhn, this PR is ready for tech. review. There’s also a required PR with this one, could you review it as well ?
👉 https://github.com/odoo/odoo/pull/89600

**Review by alexkuhn — CHANGES_REQUESTED** [review]
LGTM, I would squash commit together because they make sense together.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+

**Comment by alexkuhn** [comment]
@robodoo retry

**Pull request closed without merging**